scripts/irods/database_connect.py

```diff
@@ -197,23 +194,41 @@ def get_database_connection(irods_config):
 
     try:
         if irods_config.catalog_database_type == 'cockroachdb':
-            return pypyodbc.connect(connection_string.encode('ascii'), ansi=True, autocommit = True)
-        else:
-            return pypyodbc.connect(connection_string.encode('ascii'), ansi=True)
-    except pypyodbc.Error as e:
+            return pyodbc.connect(connection_string, ansi=True, autocommit = True)
+
+        if irods_config.catalog_database_type == 'postgres': # TODO This may be required for MySQL.
+            conn = pyodbc.connect(connection_string, ansi=True)
+
+            # The following options must be set on the connection for PostgreSQL 12.
+            # TODO This needs to be tested against a PostgreSQL 10+ database.
+            # See the following for why these are needed:
+            #
+            #   - https://github.com/mkleehammer/pyodbc/issues/169#issuecomment-268665172
+            #   - https://github.com/mkleehammer/pyodbc/issues/169#issuecomment-591979062
+            #   - https://github.com/mkleehammer/pyodbc/issues/169#issuecomment-591994201
+            #
+            conn.setencoding(encoding='utf-8')
+            conn.setdecoding(pyodbc.SQL_CHAR,  encoding='utf-8')
+            conn.setdecoding(pyodbc.SQL_WCHAR, encoding='utf-8')
```

Comment:
this is not needed if the unicode-enabled driver is used.

pyodbc must be used instead of pypyodbc because pypyodbc does not confirm to support python 3 and it does not provide functions for setting the encoding/decoding on SQL operations.

Reply:
i'm also concerned that database code written in C/C++ may not work. running this through the testing environment will help us determine what needs to happen in that case.

Reply:
this situation may be due to locale used by the container and postgres.

standing up iRODS against a properly configured UTF-8 postgres database or official postgres container may remove the need for these lines.